### ISSUE TITLE
Wrong file mode/permission when opening/creating a new file [Apple Silicon, arm64, aarch64]

### ISSUE THREAD

**Issue by 545ch4** [jruby/jruby#6983]
**Environment Information**
* `jruby 9.4.0.0-SNAPSHOT (3.0.2) 2022-01-04 5134c73021 OpenJDK 64-Bit Server VM 17.0.1+1 on 17.0.1+1 +jit [darwin-aarch64]`
* `Darwin xxx.local 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000 arm64`
* `openjdk 17.0.1 2021-10-19`

**Expected Behavior**
```
$ ruby -e "f = File.open('/tmp/test.txt', 'w'); f.puts 'Yeah!'; f.close"
$ stat /tmp/test.txt
16777232 4749651 -rw-r--r-- 1 johndoe wheel 0 6 "Jan  4 14:06:05 2022" "Jan  4 14:06:05 2022" "Jan  4 14:06:05 2022" "Jan  4 14:06:05 2022" 4096 8 0 /tmp/test.txt
```

**Actual Behavior**
```
$ jruby -e "f = File.open('/tmp/test.txt', 'w'); f.puts 'Yeah!'; f.close"
$ stat /tmp/test.txt
16777232 4751220 ---------x 1 johndoe wheel 0 6 "Jan  4 14:21:16 2022" "Jan  4 14:21:16 2022" "Jan  4 14:21:16 2022" "Jan  4 14:21:16 2022" 4096 8 0 /tmp/test.txt
```

**Comment by headius** [comment]
This is another case failing due to varargs support on M1. The `File.open` call eventually bottoms out in the native `open` call, which uses varargs for anything past the filename and mode. Since by default we try to set the file mode to 0666, the permission gets garbled on the way into the call.

**Comment by headius** [comment]
I'm going to prioritize M1 support this week and hopefully find a path forward.

**Comment by 545ch4** [comment]
Thank you @headius !

Meanwhile I was further following the backtrace to https://github.com/jnr/jnr-posix and found, that some tests are failing and especially `jnr.posix.FileStatTest` indicates, that we are on the right spot here...

**Comment by headius** [comment]
@545ch4 Yeah that is pretty much the last mile for getting this support done, and it all hinges on getting varargs working on M1. Once we have that I think we can be 100% all the way up.

**Comment by 545ch4** [comment]
@headius Yes, tests at https://github.com/jnr/jnr-ffi/blob/master/src/test/java/jnr/ffi/VarargsTest.java failing seriously: "The forked VM terminated without properly saying goodbye"... Do you need any HotSpot VM crash dump or other outputs from those tests?

**Comment by headius** [comment]
Fixed by #6985 for 9.2 and #6986 for 9.3 and beyond.
